packages/client/lib/net/protocol/ethprotocol.ts

```diff
@@ -103,11 +103,13 @@ export class EthProtocol extends Protocol {
         }
         return serializedTxs
       },
-      decode: ([txs]: [Buffer[]]) => {
-        // TODO: add proper Common instance (problem: service not accessible)
-        //const common = this.config.chainCommon.copy()
-        //common.setHardforkByBlockNumber(this.config.syncTargetHeight, this.chain.headers.td)
-        return txs.map((txData) => TransactionFactory.fromBlockBodyData(txData))
+      decode: (txs: Buffer[]) => {
+        const common = this.config.chainCommon.copy()
+        common.setHardforkByBlockNumber(
```

Comment:
i don't think we need to set hardfork by number  here, chainCommon should always have updated hf as its updated whenever putblock is done by client

Reply:
Don't we want to ensure we don't try to instantiate a transaction with a default hardfork that could be invalid for that transaction type (e.g. we don't instantiate a 1559 tx if the current hardfork is < london)?  In theory this should never happen but it seems like it's worth validating that when instantiating the transaction, right?

Reply:
We should definitely add in the hardfork by number. The reason is this: if we have a HF at height `A`, and the chain is at height `A-1`, then the txs should be validated against `A` (which is thus the HF). This could invalidate some txs, or actually validate other txs! (For instance if the HF creates a new Tx type!)

Reply:
Hmm, I'm not really sure what to do here then.  We don't store the current height of the chain anywhere else other than this `syncTargetHeight` which isn't guaranteed to be defined and I can't call any async methods to retrieve the latest header here since these encode/decode helpers are synchronous.  I would think we would want to accept transactions as long as the txPool is running and just take our best guess at a block height based on the syncTarget (or just 0 since we don't know otherwise).  I could also just take whatever the current hardfork is set in common and use that if we don't know the block number maybe?

Reply:
this.headers.latest.number should be fine, but `chainCommon` should be set at the correct hf corresponding to that, thats my point, no need to set it further since copy will copy its set hardfork

Reply:
Yeah, I would also think that a "best guess" should be appropriate here, since - technically - these txs do not have *any HF at all* yet since they are not bound to a block yet. So there is just no correct td (e.g.) at this point to assign.

So this should be ok (?) as long as it is assured that the common HF is eventually re-assigned/re-set when a tx then eventually make it in a block through the block builder?

Reply:
this could be ok if headers.td is paired with headers.latest.number else we might get td invalidation errors, which as per my understanding is the common already set in chainCommon.